app/views/errors/error.scala.html

```diff
@@ -16,12 +16,15 @@
 
 @import config.ApplicationConfig
 
-@(title: String, heading: String, message: String)(implicit request: RequestHeader, messages: Messages, config: ApplicationConfig)
+@this(mainLayout: views.html.mainLayout)
 
 
-    @main_template(title = title, topNavigation = false) {
+@(title: String, heading: String, message: String)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
```

Comment:
In the templates, we have a mixture of 
`(implicit request: RequestHeader,...`
and
`(implicit request: Request[_],...`

is it better we standardise and stick to one (RequestHeader)?

Reply:
every Request extends a RequestHeader. In ErrorHandler we have to override methods that have RequestHeader in the signature and I can't magic up a Request from that. Had to update to RequestHeader in the places where it didn't compile. There may still be a few where a Request is asked for, but that can be updated to RequestHeader...